website/src/components/visualisation/Resources/LineGraph.tsx

```diff
@@ -0,0 +1,147 @@
+                                        cy={5}
+                                        type="circle"
+                                        size={25}
+                                        fill={"#FFF"}
+                                    />
+                                )}
+                            </Surface>
+                            <span>{dataKey}</span>
+                        </span>
+                    )
+                })}
+            </div>
+        );
+    };
+
+    getSeasonEnds () {
+        var seasonEnds: any[] = [];
+        let i;
+        for(i in this.state.chartData.GameStates) {
+            if(this.state.chartData.GameStates[i].Environment.LastDisasterReport.Magnitude !== 0){
+                seasonEnds.push(this.state.chartData.GameStates[i].Turn -1)
+            }
+        }
+        return seasonEnds;
+    };
+
+    render() {
+        return (
+            <ResponsiveContainer height={330} width = "95%">
+                <LineChart data={this.state.chartData.GameStates}
+                    margin={{ top: 5, right: 30, left: 20, bottom: 5 }}>
+                    {_.toPairs(this.state.lineColours)
+                    .filter(pair => !_.includes(this.state.disabled, pair[0]))
+                    .map(pair => (
+                    <Line
+                    name={pair[0]}
+                    type="monotone"
+                    dataKey={this.state.datapaths[pair[0]]}
+                    stroke={pair[1]}
+                    />
+                    ))}
+                    <CartesianGrid strokeDasharray="3 3" />
+                    <XAxis dataKey="Turn" height={50} label={{ value: 'Turn', angle: 0, position: 'insideBottom', dy: -8}} />
+                    <YAxis label={{ value: 'Resources', angle: -90, position: 'insideLeft' }}/>
+                    <Tooltip />
+                    {this.getSeasonEnds().map((seasonEnd) => 
+                        (
+                        <ReferenceLine
+                            x={seasonEnd}
+                            label="Season End"
+                            stroke="red"
+                            strokeDasharray="3 3"
+                        />
+                    ))}
+                    <Legend 
+                        verticalAlign="top" 
+                        align="center"
+                        height={20} 
+                        wrapperStyle={{top: 0, left: 25, right: 0, width: 'auto'}}
+                        payload={_.toPairs(this.state.lineColours).map(pair => ({
+                            dataKey: pair[0], 
+                            colour: pair[1],
+                        }))}
+                        
+                        content={this.renderCustomizedLegend}
+                    />
+                    <Brush dataKey='Turn' height={28} stroke="#2fa1c6"/>
+                </LineChart>
+            </ResponsiveContainer >
+        )
+    };
+}
+
+
+export default LineRechartComponent;
+
+
+
```

Comment:
if you're using vscode `cmd+shift+f` should autoformat and remove these lines

Reply:
@ahm116 might be `ctrl (command) + shift + i`, I think lh has a custom keybind.

Additionally, some autoformatters (prettier) think that rogue newlines are user-intended and leave them alone. It's good to check manually for this. In general for typescript, 1 line between every code block. New classes are usually in a separate file, so no need to do the "2 lines between classes, 1 between functions" thing that python does.